internal/reloader/handler.go

```diff
@@ -0,0 +1,42 @@
+package reloader
+
+import (
+	"net/http"
+	"sync/atomic"
+)
+
+// AtomicHandler provides an atomic http.Handler implementation, allowing safe
+// handler replacement at runtime. AtomicHandler must be initialized with a call
+// to NewAtomicHandler. It will never panic and is safe for concurrent use.
+type AtomicHandler struct {
+	val atomic.Value
+}
```

Comment:
Is this wrapping really necessary? What do we get other than some minor type safety by just wrapping `atomic.Value`?

Reply:
It is necessary in order to implement http.Handler.